src/redemptions/redemption.service.ts

```diff
@@ -78,13 +82,21 @@ export class RedemptionService {
         idDetails,
       };
     }
+    const repeatedFace = await this.repeatedFaceCheck(transactionStatus);
```

Comment:
I'm still figuring out exact recommendations, but the duplication of the liveness/similarity/etc checks in this function and in `repeatedFaceCheck` strikes me as a code smell